articles/python-database-programming-sqlite/index.md

```diff
@@ -0,0 +1,216 @@
+```
+
+Treat the cursor as an iterator:
+
+```python
+for row in cur.execute('''SELECT * FROM STUDENTS'''):
+    print(row)
+```
+
+The result will be:
+
+```python
+(1, 'Ahmad', 20)
+(2, 'James', 22)
+(3, 'Eva', 19)
+```
+
+The SELECT statement is the most important and complex one in SQLite. Here's what you can use to make it a more specific statement:
+- Use **ORDER BY** clause to sort the result set.
+- Use **DISTINCT** clause to delete the duplicate rows in the results.
+- Use **WHERE** clause to make a condition while fetching the data from tables.
+- Use **GROUP BY** to gather data from multiple records and group the results by columns.
+- Use the **HAVING** with **GROUP BY** clause in order to filter the result based on a condition.
+
+Now let's create a complex example to get a better idea:
+
+```python
+cur.execute('''SELECT NAME FROM STUDENTS WHERE ID<3 ORDER BY AGE ASC''')  
+result = cur.fetchall()
+print(result)
+```
+
+The result will be:
+
+```python
+[('James',), ('Ahmad',)]
+```
+
+In that example, we are searching for names in table **students** where the **ID** is less than 3, and we used **ASC** to sort the column in ascending order(from the lowest value to the highest one)
+
+Also, you can use **DESC** to sort the column in descending order(from the highest value to the lowest one).
+
+### UPDATE
+Updating records in SQLite is very simple. we can update the **AGE**  of the **Students** with an id of 2 by:
+
+```python
+Update_Students ='''
+                 Update students
+                 set AGE=35
+                 WHERE ID=2
+                 '''
+cur.execute(Update_Students)
+```
+
+Now, if we execute the SELECT query(SELECT * FROM STUDENTS), we can see the following result:
+
+```python
+[(1, 'Ahmad', 20), (2, 'James', 35), (3, 'Eva', 19)]
+```
+
+### DELETE
+So far, we covered how to **insert** new data into a table, **update** an existing row, and **SELECT** data from a table. Sometimes, you need to delete data from a table.
+
+Let's go over how to delete data.
+
+As an example, try to delete the student with an id of 3:
+
+```python
+Delete_Student = '''DELETE FROM STUDENTS WHERE id = 3'''
+cur.execute(Delete_Student)
+```
+
+Now, if you take a look at **STUDENTS** table, you’ll see that the third student has been deleted.
+
+```python
+[(1, 'Ahmad', 20), (2, 'James', 35)]
+```
+
+In the end, we have to commit our changes in the database. Not calling this method, will make anything you did after the last commit() not visible for the other database connection. So you have to make sure that you used this method before closing the connection.
+
+```python
+connection.commit()
+```
+
+Then we have to close the connection to the database by using:
+
+```python
+connection.close()
+```
+
+When the connection is closed then a ROLLBACK will happen in the next connection to the database to any outstanding transaction. (The ROLLBACK statement cancels transaction that did any changes to the database).
```

Comment:
This statement is a little confusing, you should revise to make it a bit more clear that outstanding transactions == un-committed changes. Also, the parenthesis should be removed from the ROLLBACK clarification sentence.